subsquid/src/dbHelper.ts

```diff
@@ -481,27 +480,80 @@ export async function getOrCreateHistoricalAssetPrice(
 }
 
 /**
- * Gets current prices from Coingecko
+ * Gets current prices from DB or Coingecko
  * @param ctx
  */
 export async function getCurrentAssetPrices(
   ctx: EventHandlerContext<Store> | EntityManager
 ): Promise<Record<string, number> | undefined> {
-  const endpoint = "https://api.coingecko.com/api/v3/simple/price?ids=tether%2Ckusama&vs_currencies=usd";
+  const isRepository = ctx instanceof EntityManager;
+
+  let currentPrices: Record<string, number> = {};
+
+  const now = new Date();
+  // Round time to the nearest minute
+  const date = new Date(now.getFullYear(), now.getMonth(), now.getDate(), now.getHours(), now.getMinutes());
+
+  const assetIds = ["1", "4", "130"];
 
-  const res = await fetch(endpoint);
-  if (!res.ok) {
-    throw new Error("Failed to fetch prices from coingecko");
+  let missingPrices = false;
+
+  // Check if all prices are in DB
+  for (const assetId of assetIds) {
+    // Search for price in DB
+    const where = {
+      assetId,
+      timestamp: date
+    };
+    const assetPrice = isRepository
+      ? await ctx.getRepository(HistoricalAssetPrice).findOne({ where })
+      : await ctx.store.findOne(HistoricalAssetPrice, { where });
+
+    // If any price is missing, flag it and break
+    if (assetPrice?.price === undefined) {
+      missingPrices = true;
+      break;
+    } else {
+      currentPrices[assetId] = assetPrice.price;
+    }
   }
-  const json: { kusama: { usd: number }; tether: { usd: number } } = await res.json();
 
-  const picaKsmSpotPrice = await getSpotPrice(ctx, "1", "4", "2");
+  // If any price is missing, get all from Coingecko and update DB
+  if (missingPrices) {
+    const endpoint = "https://api.coingecko.com/api/v3/simple/price?ids=tether%2Ckusama&vs_currencies=usd";
```

Comment:
probably not the best idea to hardcode this?

Reply:
Done @fl-y ✅